staging/src/k8s.io/apiserver/pkg/storage/etcd3/lease_manager.go

```diff
@@ -57,24 +61,26 @@ type leaseManager struct {
 	// The period of time in seconds and percent of TTL that each lease is
 	// reused. The minimum of them is used to avoid unreasonably large
 	// numbers.
-	leaseReuseDurationSeconds int64
-	leaseReuseDurationPercent float64
-	leaseAttachedObjectCount  int64
+	leaseReuseDurationSeconds   int64
+	leaseReuseDurationPercent   float64
+	leaseMaxAttachedObjectCount int64
+	leaseAttachedObjectCount    int64
 }
 
 // newDefaultLeaseManager creates a new lease manager using default setting.
 func newDefaultLeaseManager(client *clientv3.Client, config LeaseManagerConfig) *leaseManager {
-	return newLeaseManager(client, config.ReuseDurationSeconds, 0.05)
+	return newLeaseManager(client, config.ReuseDurationSeconds, 0.05, config.MaxObjectCount)
```

Comment:
If config.MaxObjectCount is <= 0, I think you should either choose a reasonable value or panic. Otherwise callers we didn't notice (I'm mostly thinking of tests) might end up with zero as the value, which definitely seems wrong.